src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -177,18 +177,21 @@ define(function (require) {
         return this.intervalName && _.find(intervals, { name: this.intervalName });
       };
 
-      self.toIndexList = function (start, stop, sortDirection) {
```

Comment:
Rather than adding complexity to this function by making it polymorphic, how about we move all of this logic into a new `toDetailedIndexList` function, and then modify the existing `toIndexList` to do something like:

```js
self.toIndexList = function (start, stop, sortDirection) {
  return this.toDetailedIndexList()
  .then(indices => return _.pluck(indices, 'index'));
};
```